RegressionTests.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE OverloadedStrings #-}
+module RegressionTests where
+
+import Database.HDBC.Sqlite3
+import Database.HDBC
+import System.IO
+import System.Environment
+import System.Exit
+import System.Process
+import Data.Aeson
+import Data.Aeson.Types
+import Data.Maybe
+import qualified Data.ByteString.Lazy.UTF8 as BLU
+import qualified Data.HashMap.Strict as HM
+import Data.Text (Text)
+import qualified Data.Text.IO as TIO
+import Data.List
+
+status_list :: [Text]
+status_list = ["", "syntaxerror", "notequiv", "similar", "correct", "expected", "buggy"]
```

Comment:
Why is `"buggy"` on the right? Suppose the old tutor gives `"correct"` for one example, and the new tutor gives `"buggy"`: I would consider that a regression, because you assume the old tutor is the correct reference implementation.

Reply:
The status `correct` means the tutor found improvement, but was unsure what. The status `expected` means that the tutor knew what was corrected. I agree with you on your point though, so I'll move buggy down, but only to before `expected` as there can be cases where the old tutor thought a correct refactoring step was made but unsure what, while it was a buggy rule. For example:

old tutor:
`count = count + 1` ~> `count++` (expected)
`count = count + 1` ~> `count = count++` (correct)

new tutor:
`count = count + 1` ~> `count++` (expected)
`count = count + 1` ~> `count = count++` (buggy)